contribs/gmf/src/layertree/component.js

```diff
@@ -759,21 +760,69 @@ Controller.prototype.afterReorder = function() {
 };
 
 
+/**
+ * @param {import('gmf/themes.js').GmfGroup} node Layer tree node to tag popup identifier.
+ */
+Controller.prototype.tagPopup = function(node) {
+  const uid = getUid(node);
+
+  // Find the random id associated with the new popup being opened.
+  const popupId = $(`span.extra-actions#popup-id-${uid}`).attr('aria-describedby');
+  if (!node.popupId) {
+    node.popupId = popupId;
+  } else {
+    delete node.popupId;
+  }
+};
+
+
 /**
  * @param {import('gmf/themes.js').GmfGroup} node Layer tree node to remove.
  */
 Controller.prototype.removeNode = function(node) {
+  this.parseTreeNodes(node);
   this.gmfTreeManager_.removeGroup(node);
 };
 
 
-/**
- */
 Controller.prototype.removeAllNodes = function() {
+  const nodes = this.root.children;
+  nodes.forEach(node => {
+    this.parseTreeNodes(node);
+  });
```

Comment:
You can do `this.parseTreeNodes(this.root)`, right?

Reply:
Right, forgot to change that after doing my recursive function. Now fixed.